pkg/ceph/osd/device.go

```diff
@@ -422,18 +471,18 @@ func getStoreSettings(config *osdConfig, storeConfig StoreConfig) (map[string]st
 	return settings, nil
 }
 
-func writeConfigFile(config *osdConfig, context *clusterd.Context, cluster *mon.ClusterInfo, storeConfig StoreConfig) error {
+func writeConfigFile(config *osdConfig, context *clusterd.Context, cluster *mon.ClusterInfo) error {
 	cephConfig := mon.CreateDefaultCephConfig(context, cluster, config.rootPath, isBluestore(config))
 
-	if !isBluestore(config) {
+	if config.dir || isFilestoreDevice(config) {
```

Comment:
isn't this the same as `if isFilestorage(config)`?

Reply:
no, this conditional needs to cover the following cases:
* bluestore directory
* filestore directory
* filestore device